judo-ui-react/src/main/resources/actor/src/layout/BottomMenu/index.tsx.hbs

```diff
@@ -0,0 +1,357 @@
+          />
+        );
+      case 'collapse':
+        return (
+          <NavCollapse
+            key={item.id}
+            menu={item}
+            level={2}
+            render={true}
+            placement="auto"
+            arrow={false}
+            mouseOver={false}
+          />
+        );
+      case 'item':
+        return <NavItem key={item.id} item={item} level={2} />;
+      default:
+        return (
+          <Typography key={item.id} variant="h6" color="error" align="center">
+            Menu Items Error
+          </Typography>
+        );
+    }
+  };
+
+  const handlePopupOpen = (event: React.MouseEvent<HTMLElement>, item: NavItemType) => {
+    setMouseX(event.clientX);
+    setMouseY(event.clientY);
+    setSelectedBottomMenuItem(item);
+    setSelectedBottomOther(false);
+    setSubMenuItems(
+      item.children?.map((menu) => {
+        return popupMenuItemAction(menu);
+      }),
+    );
+    setOpen(true);
+  };
+
+  const handleOtherPopupOpen = (event: React.MouseEvent<HTMLElement>) => {
+    setMouseX(event.clientX);
+    setMouseY(event.clientY);
+    setSelectedBottomMenuItem(undefined);
+    setSelectedBottomOther(true);
+    setSubMenuItems([
+      ...bottomMenuItems.slice(3, lastItemIndex + 1).map((item) => {
+        return popupMenuItemAction(item);
+      }),
+    ]);
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+    setSubMenuItems(undefined);
+    setSelectedBottomMenuItem(undefined);
+    setSelectedBottomOther(false);
+  };
+
+  const [mouseX, setMouseX] = useState(0);
+  const [mouseY, setMouseY] = useState(0);
+
+  const itemAction = (item: NavItemType) => {
+    switch (item.type) {
+      case 'group':
+        return (
+          <BottomNavigationAction
+            aria-describedby="submenu"
+            showLabel={true}
+            key={item.id + 'NavAction'}
+            id={item.id + 'NavAction'}
+            onClick={(event) => {
+              handlePopupOpen(event, item);
+            } }
+            icon={<MdiIcon path={item.icon ?? 'dots-vertical'} />}
+            label={t(`menuTree.${item.title}`, { defaultValue: item.title })}
+            content=""
+          ></BottomNavigationAction>
+        );
+      case 'item':
+        return (
+          <BottomNavigationAction
+            key={item.id + 'NavAction'}
+            onClick={(event) => handlePerform(event, item)}
+            icon={<MdiIcon path={item.icon ?? 'dots-vertical'} />}
+            label={t(`menuTree.${item.title}`, { defaultValue: item.title })}
+          />
+        );
+      default:
+        return (
+          <Typography key={item.id} variant="h6" color="error" align="center">
+            Invalid item type: {item.type}!
+          </Typography>
+        );
+    }
+  };
+
+  const otherAction = () => {
+    return (
+      <BottomNavigationAction
+        aria-describedby="submenu"
+        showLabel={true}
+        key={'_OtherNavAction'}
```

Comment:
We don't need this if a component is not an immediate child of a `.map()` iteration.